apps/docs/content/guides/local-development/declaring-database-schemas.mdx

```diff
@@ -0,0 +1,231 @@
+
+## Schema migrations
+
+Schema migrations are SQL statements written in Data Definition Language. They are [versioned](/docs/guides/deployment/database-migrations) in your `supabase/migrations` directory to ensure schema consistency between local and remote environments.
+
+### Declaring your schema
+
+<StepHikeCompact>
+
+  <StepHikeCompact.Step step={1}>
+    <StepHikeCompact.Details title="Create your first schema file">
+      Create a SQL file in `supabase/schemas` directory that defines an `employees` table.
+    </StepHikeCompact.Details>
+
+    <StepHikeCompact.Code>
+
+```sql supabase/schemas/employees.sql
+create table "employees" (
+  "id" integer not null,
+  "name" text
+);
+```
+
+    </StepHikeCompact.Code>
+
+  </StepHikeCompact.Step>
+</StepHikeCompact>
+
+<StepHikeCompact>
+
+  <StepHikeCompact.Step step={2}>
+    <StepHikeCompact.Details title="Generate a migration file">
+      Generate a migration file by diffing against your declared schema.
+    </StepHikeCompact.Details>
+
+    <StepHikeCompact.Code>
+
+```bash Terminal
+supabase db diff -f employees_table
+```
+
+    </StepHikeCompact.Code>
+
+  </StepHikeCompact.Step>
+</StepHikeCompact>
+
+<Admonition type="info">
+
+Make sure your local database is stopped before diffing your schema.
+
+</Admonition>
+
+<StepHikeCompact>
+
+  <StepHikeCompact.Step step={3}>
+    <StepHikeCompact.Details title="Start the local database">
+      The new migration file will be used when starting the database locally so you can validate your schema using the local Dashboard.
+    </StepHikeCompact.Details>
+
+    <StepHikeCompact.Code>
+
+```bash Terminal
+supabase start
+```
+
+    </StepHikeCompact.Code>
+
+  </StepHikeCompact.Step>
+</StepHikeCompact>
+
+### Updating your schema
+
+<StepHikeCompact>
+
+  <StepHikeCompact.Step step={1}>
+    <StepHikeCompact.Details title="Stop the local database">
+      Before updating your schema files, stop the local development environment.
+    </StepHikeCompact.Details>
+
+    <StepHikeCompact.Code>
+
+```bash Terminal
+supabase stop
+```
+
+    </StepHikeCompact.Code>
+
+  </StepHikeCompact.Step>
+</StepHikeCompact>
+
+<StepHikeCompact>
+
+  <StepHikeCompact.Step step={2}>
+    <StepHikeCompact.Details title="Add a new column">
+      Edit `supabase/schemas/employees.sql` file to add a new column to `employees` table.
+    </StepHikeCompact.Details>
+
+    <StepHikeCompact.Code>
+
+```sql supabase/schemas/employees.sql
+create table "employees" (
+  "id" integer not null,
+  "name" text,
+  "age" smallint not null
+);
+```
+
+    </StepHikeCompact.Code>
+
+  </StepHikeCompact.Step>
+</StepHikeCompact>
+
+<Admonition type="info">
+
+Some entities like views and enums depend columns to be declared in a specific order. To avoid messy diffs, always add new columns by appending to the end of the table.
```

Comment:
```suggestion
Some entities like views and enums expect columns to be declared in a specific order. To avoid messy diffs, always add new columns by appending to the end of the table.
```